File: test/ch21-benchmarks/bench-011-nested-loops.jl
# BENCH-011: Nested Loops (1000x1000 iterations) - Julia

function nested_loops(outer, inner)
    total = 0
    for i in 0:outer-1
        for j in 0:inner-1
            total += i * j
        end
    end
    total
end

result = nested_loops(1000, 1000)
# Expected: 249500250000

File: test/ch21-benchmarks/bench-011-nested-loops.ts
// BENCH-011: Nested Loops (1000x1000 iterations) - TypeScript/Deno

function nestedLoops(outer: number, inner: number): number {
    let total = 0;
    for (let i = 0; i < outer; i++) {
        for (let j = 0; j < inner; j++) {
            total += i * j;
        }
    }
    return total;
}

const result = nestedLoops(1000, 1000);
// Expected: 249500250000
